lib/ipython_autoimport.py

```diff
@@ -57,12 +59,15 @@ def visit_ImportFrom(self, node):
 
     for _, _, entry in (
             ipython.history_manager.get_tail(
-                ipython.history_load_length, raw=False)):
-        try:
-            parsed = ast.parse(entry)
-        except SyntaxError:
-            continue
-        Visitor().visit(parsed)
+                ipython.history_load_length, raw=True)):
+        if entry.startswith("%autoimport -clear "):
+            import_cache.pop(entry[len("%autoimport -clear "):], None)
```

Comment:
I think this is slightly brittle (e.g. would break if there's two spaces after `--clear`, or if one used `-c`); perhaps better would be to send any line starting with `%autoimport ` to the same argument parser as `%autoimport`?